### ISSUE TITLE
functions: enable integration tests

### ISSUE THREAD

**Issue by tbpg** [GoogleCloudPlatform/golang-samples#1381]
The Cloud Functions integration tests rely on the `BASE_URL` environment variable for getting the URL of deployed Cloud Functions. Those tests are always skipped.

For example:

https://github.com/GoogleCloudPlatform/golang-samples/blob/98312c1ac4900a276d3ff2af286f6f9e0fe5af7b/functions/http/cors_auth_test.go#L72-L80

We need to figure out a good way to deploy the functions before we can set `BASE_URL` to anything.

We can deploy the functions privately and follow #1380 to query them.

**Comment by codyoss** [comment]
cc @codyoss

**Comment by ace-n** [comment]
You should be able to deploy these in a dedicated testing project, and use that info in the `BASE_URL`

**Comment by grayside** [comment]
I imagine we need something like https://github.com/GoogleCloudPlatform/golang-samples/tree/master/internal/cloudrunci, though I am currently considering a different strategy.

**Comment by ace-n** [comment]
Ping on this - any updates? 🙂

**Comment by grayside** [comment]
No, but we should talk as this falls into our test planning project.

**Comment by grayside** [comment]
I've started building something for this, and filed #1690 as a blocking feature request to enable this issue to move forward.

**Comment by tbpg** [comment]
Closing as a duplicate of #1690 (both issues have relevant discussions).

**Comment by grayside** [comment]
Note the distinction is #1690 will not include enabling testing for all functions. It will target a base pattern with 1-2 examples.
